packages/e2e-tests/specs/plugins/nonce.test.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	activatePlugin,
+	createNewPost,
+	deactivatePlugin,
+	saveDraft,
+} from '@wordpress/e2e-test-utils';
+
+describe( 'Nonce', () => {
+	beforeAll( async () => {
+		await activatePlugin( 'gutenberg-test-plugin-nonce' );
+	} );
+
+	afterAll( async () => {
+		await deactivatePlugin( 'gutenberg-test-plugin-nonce' );
+	} );
+
+	beforeEach( async () => {
+		await createNewPost();
+	} );
+
+	it( 'should refresh when expired', async () => {
+		await page.keyboard.press( 'Enter' );
+		// eslint-disable-next-line no-restricted-syntax
```

Comment:
Can you justify this?

Reply:
We have an eslint rule to discourage `page.waitFor`. Is there an alternative?

Reply:
> We have an eslint rule to discourage `page.waitFor`. Is there an alternative?

Most times people feel the need to use it, it's usually symptomatic of an underlying issue, or at least they should be more concerned with awaiting the presence of some element which could guarantee (rather than hope based on timeout) a satisfied condition for proceeding.

In your case, I think since you're explicitly awaiting the nonce timeout, it's fine to use it. I might suggest to include a corresponding comment explaining why you're waiting though.

Related: #16795

Does the timeout need to be so long? Since this will cause real-time delay on the test runtime, I might suggest it be as low as possible (1s?).

Reply:
Yeah, I think 1s should be ok too.